Sources/MongoSwift/MongoError.swift

```diff
@@ -169,6 +172,95 @@ internal func parseMongocError(error: bson_error_t, errorLabels: [String]? = nil
     }
 }
 
+/// Internal function used to get an appropriate error from a server reply to a command.
+internal func getErrorFromReply(
```

Comment:
this might be a bit more readable if we broke out the bulk write and write error portions into separate methods.

Reply:
done